src/m365/teams/commands/app/app-list.ts

```diff
@@ -47,100 +42,38 @@ class TeamsAppListCommand extends GraphCommand {
       }
     );
   }
 
   #initValidators(): void {
     this.validators.push(
       async (args: CommandArgs) => {
-        if (args.options.teamId && args.options.teamName) {
-          return 'Specify either teamId or teamName, but not both.';
-        }
-
-        if (args.options.teamId && !validation.isValidGuid(args.options.teamId)) {
-          return `${args.options.teamId} is not a valid GUID`;
+        if (args.options.distributionMethod &&
+          TeamsAppListCommand.allowedDistributionMethods.indexOf(args.options.distributionMethod) < 0) {
+          return `'${args.options.distributionMethod}' is not a valid distribution method. Allowed distribution methods are ${TeamsAppListCommand.allowedDistributionMethods.join(', ')}`;
         }
 
         return true;
       }
     );
   }
 
-  private getTeamId(args: CommandArgs): Promise<string> {
-    if (args.options.teamId) {
-      return Promise.resolve(args.options.teamId);
-    }
-
-    return aadGroup
-      .getGroupByDisplayName(args.options.teamName!)
-      .then(group => {
-        if ((group as ExtendedGroup).resourceProvisioningOptions.indexOf('Team') === -1) {
-          return Promise.reject(`The specified team does not exist in the Microsoft Teams`);
-        }
-
-        return group.id!;
-      });
-  }
-
-  private getEndpointUrl(args: CommandArgs): Promise<string> {
-    return new Promise<string>((resolve: (endpoint: string) => void, reject: (error: string) => void): void => {
-      if (args.options.teamId || args.options.teamName) {
-        this
-          .getTeamId(args)
-          .then((teamId: string): void => {
-            let endpoint: string = `${this.resource}/v1.0/teams/${formatting.encodeQueryParameter(teamId)}/installedApps?$expand=teamsApp`;
-
-            if (!args.options.all) {
-              endpoint += `&$filter=teamsApp/distributionMethod eq 'organization'`;
-            }
-
-            return resolve(endpoint);
-          })
-          .catch((err: any) => {
-            reject(err);
-          });
-      }
-      else {
-        let endpoint: string = `${this.resource}/v1.0/appCatalogs/teamsApps`;
-
-        if (!args.options.all) {
-          endpoint += `?$filter=distributionMethod eq 'organization'`;
-        }
-
-        return resolve(endpoint);
-      }
-    });
-  }
-
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     try {
-      const endpoint = await this.getEndpointUrl(args);
-      const items = await odata.getAllItems<TeamsApp>(endpoint);
-
-      if (args.options.teamId || args.options.teamName) {
-        items.forEach(t => {
-          t.displayName = (t as any).teamsApp.displayName;
-          t.distributionMethod = (t as any).teamsApp.distributionMethod;
-        });
-      }
+      const requestUrl: string = `${this.resource}/v1.0/appCatalogs/teamsApps${args.options.distributionMethod ? `?$filter=distributionMethod eq '${args.options.distributionMethod}'` : ''}`;
```

Comment:
Instead of the inline if, could we do something like:

```
let requestUrl = 'https://...'

if (args.options.distributionMethod) {
   requestUrl += `$filter=...`
}
```
This improves the readability a lot.